### PR TITLE
Update sacrebleu to version 1.2

### PR DESCRIPTION
**Pull request by mjpost**
- added the chrF metric (`-m chrf` or `-m bleu chrf` for both)
     See 'CHRF: character n-gram F-score for automatic MT evaluation' by Maja Popovic (WMT 2015)
     [http://www.statmt.org/wmt15/pdf/WMT49.pdf]
   - added IWSLT 2017 test and tuning sets for DE, FR, and ZH
     (Thanks to Mauro Cettolo and Marcello Federico).
   - added `--cite` to produce the citation for easy inclusion in papers
   - added `--input` (`-i`) to set input to a file instead of STDIN
   - removed accent mark (on private suggestion)

(description of the change)

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [ ] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./pre-commit.sh` or manual run of pylint & mypy)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [ ] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: contrib/sacrebleu/sacrebleu.py [hunk 1]
```diff
@@ -146,15 +157,15 @@
...
 import unicodedata
+import numpy
 
 from collections import Counter, namedtuple
 from itertools import zip_longest
-from typing import List
+from typing import List, Iterable, Tuple
 
-VERSION = '1.1.7'
+VERSION = '1.1.8'
```
Comment (fhieber) [inline contrib/sacrebleu/sacrebleu.py]: Why not 1.2?

File: contrib/sacrebleu/sacrebleu.py [hunk 2]
```diff
@@ -658,7 +827,7 @@ def build_signature(args, numrefs):
     return sigstr
 
 
-def extract_ngrams(line, max_order=NGRAM_ORDER):
+def extract_ngrams(line, min_order=1, max_order=NGRAM_ORDER) -> Counter:
```
Comment (fhieber) [inline contrib/sacrebleu/sacrebleu.py]: you could generalize extract_ngrams to cover the character case as well. (either feed in a list of strings or a string (which is also indexable)).

File: contrib/sacrebleu/setup.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def get_version():
     # your project is installed. For an analysis of "install_requires" vs pip's
     # requirements files see:
     # https://packaging.python.org/en/latest/requirements.html
-    install_requires = ['typing'],
+    install_requires = ['typing', 'numpy'],
```
Comment (fhieber) [inline contrib/sacrebleu/setup.py]: does this add too many dependencies for this lightweight project?

File: sockeye/evaluate.py [hunk 4]
```diff
@@ -41,6 +40,15 @@ def raw_corpus_bleu(hypotheses: Iterable[str], references: Iterable[str], offset
...
 
+def raw_corpus_chrf(hypotheses: Iterable[str], references: Iterable[str]) -> float:
+    """
+    Simple wrapper around sacreBLEU's chrF implementation, without tokenization.
+
+    :param hypotheses: Hypotheses stream.
+    :param references: Reference stream.
+    :return: chrF score as float between 0 and 1.
+    """
+    return sacrebleu.corpus_chrf(hypotheses, references, remove_whitespace=True)
```
Comment (fhieber) [inline sockeye/evaluate.py]: lets make the default order and beta parameters explicit here: `order=sacrebleu.CHRF_ORDER` etc.

File: contrib/sacrebleu/sacrebleu.py (hunk 1 above)
Comment (mjpost) [reply contrib/sacrebleu/sacrebleu.py]: whoops

File: contrib/sacrebleu/setup.py (hunk 3 above)
Comment (mjpost) [reply contrib/sacrebleu/setup.py]: I just copied it from the old chrF implemention, but you're right that it's overkill.

File: sockeye/evaluate.py (hunk 4 above)
Comment (mjpost) [reply sockeye/evaluate.py]: Will do.

File: contrib/sacrebleu/sacrebleu.py (hunk 2 above)
Comment (mjpost) [reply contrib/sacrebleu/sacrebleu.py]: There is definitely some overlap here, but I gave up on combining them for diminishing returns. One problem is that a slice of a string is a string, but a slice of a list is a list, which can't work as a key for a Counter.

**Comment by mjpost** [comment]
@fhieber If you're okay with not merging the ngram code, then this is ready now.

**Pull request merged**